src/libs/portfolio/portfolio.ts

```diff
@@ -163,7 +163,13 @@ export class Portfolio {
     }
 
     const tokenFilter = ([error, result]: [string, TokenResult]): boolean =>
-      (result.amount > 0 || pinned!.includes(result.address)) &&
+      (result.amount > 0 ||
+        !!pinned.find((pinnedToken) => {
+          // Pinned tokens from the humanizer don't have a networkId
+          if (pinnedToken.networkId === null) return pinnedToken.address === result.address
```

Comment:
we need to change the humanizer perhaps because this could be a security risk in very weird circumstances/edge cases

Reply:
@JIOjosBG @borislav-itskov

Reply:
Here is a proposal for the future structure for humanizerMeta. 
My initial idea is to store whether it is pinned in the humanizerMeta, but there might be better and simpler solution
https://github.com/AmbireTech/ambire-app/issues/1662